unix_integration/src/unix_proto.rs

```diff
@@ -30,28 +30,34 @@ pub struct DeviceAuthorizationResponse {
     pub message: Option<String>,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub enum MFAType {
+    Code, // PAM must prompt for an authentication code
+    Poll, // PAM can immediately return. The provider will poll for an external response
+}
+
 #[derive(Serialize, Deserialize, Debug)]
 pub enum PamAuthResponse {
     Unknown,
     Success,
     Denied,
     Password,
-    DeviceAuthorizationGrant { data: DeviceAuthorizationResponse },
-    /*
+    DeviceAuthorizationGrant {
+        data: DeviceAuthorizationResponse,
+    },
     MFACode {
+        msg: String,
+        mfa_type: MFAType,
```

Comment:
Is there a reason for this to be an inner enum rather than just different variants of PamAuthResponse?